oak-search-elastic/src/main/java/org/apache/jackrabbit/oak/plugins/index/elasticsearch/index/ElasticsearchIndexWriter.java

```diff
@@ -142,7 +150,49 @@ public boolean close(long timestamp) throws IOException {
     // TODO: we need to check if the index already exists and in that case we have to figure out if there are
     // "breaking changes" in the index definition
     protected void provisionIndex() throws IOException {
-        CreateIndexRequest request = new CreateIndexRequest(indexDefinition.getRemoteIndexName());
+
+        IndicesClient indicesClient = elasticsearchConnection.getClient().indices();
+        final String indexName = indexDefinition.getRemoteIndexName();
+
+        CreateIndexRequest createIndexRequest = constructCreateIndexRequest(indexName);
+        String requestMsg = Strings.toString(createIndexRequest.toXContent(jsonBuilder(), EMPTY_PARAMS));
+        CreateIndexResponse response = indicesClient.create(createIndexRequest, RequestOptions.DEFAULT);
+
+        LOG.info("Updated settings for index {} = {}. Response acknowledged: {}",
+                indexDefinition.getRemoteIndexAlias(), requestMsg, response.isAcknowledged());
+
+
+        GetAliasesRequest getAliasesRequest = new GetAliasesRequest(indexDefinition.getRemoteIndexAlias());
+        GetAliasesResponse aliasesResponse = indicesClient.getAlias(getAliasesRequest, RequestOptions.DEFAULT);
+        Map<String, Set<AliasMetaData>> aliases = aliasesResponse.getAliases();
+        IndicesAliasesRequest indicesAliasesRequest = new IndicesAliasesRequest();
+        for (String oldIndexName : aliases.keySet()) {
+            IndicesAliasesRequest.AliasActions removeAction = new IndicesAliasesRequest.AliasActions(IndicesAliasesRequest.AliasActions.Type.REMOVE);
+            removeAction.index(oldIndexName).alias(indexDefinition.getRemoteIndexAlias());
+            indicesAliasesRequest.addAliasAction(removeAction);
+        }
+        IndicesAliasesRequest.AliasActions addAction = new IndicesAliasesRequest.AliasActions(IndicesAliasesRequest.AliasActions.Type.ADD);
+        addAction.index(indexName).alias(indexDefinition.getRemoteIndexAlias());
+        indicesAliasesRequest.addAliasAction(addAction);
+        AcknowledgedResponse updateAliasResponse = indicesClient.updateAliases(indicesAliasesRequest, RequestOptions.DEFAULT);
+        LOG.info("Updated alias {} to index {}. Response acknowledged: {}", indexDefinition.getRemoteIndexAlias(),
+                indexDefinition.getRemoteIndexName(), updateAliasResponse.isAcknowledged());
```

Comment:
This is what i was referring to in our discussion today. The acknowledgment of response `updateAliasResponse.isAcknowledged()` is just logged. The same is done with the response of create index and delete index calls. 
I was wondering if we should throw exceptions instead of just logging, in case acknowledgment is false.

Reply:
I think we should throw an `IllegalStateException` in those cases

Reply:
Makes sense. I will update the code.